app/grandchallenge/components/tasks.py

```diff
@@ -1064,38 +1051,68 @@ def validate_voxel_values(*, civ_pk):
     civ.interface._validate_voxel_values(civ.image)
 
 
-@acks_late_micro_short_task
+def update_uploadsession_and_object_status(
+    *, error_message, upload_session, object
+):
+    formatted_error = format_validation_error_message(error=error_message)
+    upload_session.status = RawImageUploadSession.FAILURE
+    upload_session.error_message = formatted_error
+    upload_session.save()
+    if hasattr(object, "error_message"):
+        object.status = object.CANCELLED
+        object.error_message = formatted_error
```

Comment:
Are you sure that these attributes exist and are database fields? Why are you not using the update status method?